### ISSUE TITLE
Nullable warnings not appearing in IDE

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#67839]
**Version Used**: 17.6 preview 3

**Steps to Reproduce**:

Can't reproduce in a new project. Can reproduce in Compilers.sln (I happened to be in LocalRewriter_Call.cs).

Paste the following inside a method in a source file:

```cs
#nullable enable
string? x = null;
x.ToString();
```

**Expected Behavior**: A warning appears on `x` in `x.ToString()`.

**Actual Behavior**: No warning appears in the IDE, but a warning does appear in the build.

**Comment by Youssef1313** [comment]
@RikkiGibson I see the warning:

![image](https://user-images.githubusercontent.com/31348972/232582270-173f7ce5-4746-4dc0-a5e2-e36178749bfe.png)

Though the problem for me is that Error List is completely empty. It says 1 warning correctly:

![image](https://user-images.githubusercontent.com/31348972/232582405-96cf7825-7ea7-4a3b-86b9-e4125bf3b71e.png)

but...

![image](https://user-images.githubusercontent.com/31348972/232582451-50375e2c-7dc1-4336-b8ee-cc4519b5b5f7.png)

![image](https://user-images.githubusercontent.com/31348972/232582501-1d9ad85b-aef9-432d-a989-247c25ffcc68.png)

![image](https://user-images.githubusercontent.com/31348972/232582539-0cd656b4-80e5-4555-90fd-367028bdd524.png)

**Comment by RikkiGibson** [comment]
What's weird for me is, and I should have mentioned: if I highlight and hit tab to mess up the formatting, I get "fix formatting" warnings with squiggles. But still no nullable warning.

**Comment by Youssef1313** [comment]
@RikkiGibson The only place I know of that disables nullable warnings is:

https://github.com/dotnet/roslyn/blob/e6817547af8b16994e31ae959ec28e4cb4922020/src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets#L85-L100

But it **shouldn't** happen for IDE live analysis.

**Comment by RikkiGibson** [comment]
Dug a little bit and noticed that the warning is in the Error List. Just no squiggle.

![image](https://user-images.githubusercontent.com/5833655/232583366-dbe54bbb-ebb2-4a83-b7c7-2d82e9e560d6.png)

**Comment by Youssef1313** [comment]
@RikkiGibson and no help link. There is a help link in my case:

![image](https://user-images.githubusercontent.com/31348972/232583627-33533d28-6fc7-4bc7-90e4-84af48151b86.png)

**Comment by Youssef1313** [comment]
![image](https://user-images.githubusercontent.com/31348972/232583952-9ab59d27-9446-4c9c-86d6-46c58688b2a4.png)

![image](https://user-images.githubusercontent.com/31348972/232583970-4fd3f6f3-1387-4189-a9a8-f60dc9844a3c.png)

Not sure if that can help investigating, but might be worth checking the "Source" of the diagnostic in error list.

**Comment by RikkiGibson** [comment]
Still seeing this in 17.6 preview 4.

**Comment by CyrusNajmabadi** [comment]
I don't repro this.
